network/network.go

```diff
@@ -138,8 +150,8 @@ func (nm *networkManager) newNetwork(nwInfo *NetworkInfo) (*network, error) {
 		nwInfo.Mode = opModeDefault
 	}
 
-	// Find the external interface for this subnet.
-	extIf := nm.findExternalInterfaceBySubnet(nwInfo.Subnets[0].Prefix.String())
+	// Find the external interface by name.
+	extIf := nm.findExternalInterfaceByName(nwInfo.MasterIfName)
```

Comment:
can we do something like:

extIf, exists := nm.ExternalInterfaces(nwInfo.MasterIfName)
if !exists {
    ...
}